python-bindings/b2_terraform/provider_tool.py

```diff
@@ -366,12 +379,27 @@ def _preprocess(self, **kwargs):
 
         server_side_encryption = kwargs.pop('server_side_encryption')
         if server_side_encryption:
-            mode = apply_or_none(EncryptionMode, server_side_encryption[0]['mode'] or None)
+            mode = server_side_encryption[0]['mode'] or None
             if mode:
-                algorithm = apply_or_none(
-                    EncryptionAlgorithm, server_side_encryption[0]['algorithm'] or None
+                customer_key = None
+                if mode != "none":
+                    algorithm = apply_or_none(
+                        EncryptionAlgorithm, server_side_encryption[0]['algorithm'] or 'AES256'
+                    )
+                    if mode == "SSE-C":
+                        key = server_side_encryption[0]['key'][0]
+                        # EncryptionKey only accepts raw bytes as keys, not base 64
+                        customer_key = EncryptionKey(
+                            secret=base64.b64decode(key['secret_b64'], validate=True),
+                            key_id=key.get('key_id'),
+                        )
+                        if len(customer_key.secret) != 32:
+                            raise Exception(f'Wrong key length ({len(customer_key.secret)})')
```

Comment:
In other parts of the code, we use `RuntimeError` which in my opinion is a better choice.